lib/active_record/typed_store/dsl.rb

```diff
@@ -1,29 +1,45 @@
+require 'active_record/typed_store/column'
 module ActiveRecord::TypedStore
-
   class DSL
+    attr_reader :columns, :coder
 
-    attr_reader :columns
-
-    def initialize(accessors=true)
-      @accessors = accessors
-      @columns = []
+    def initialize(options)
+      @coder = options.fetch(:coder, default_coder)
+      @columns = {}
       yield self
     end
 
+    def default_coder
+      ActiveRecord::Coders::YAMLColumn.new
+    end
+
     def accessors
-      @columns.select(&:accessor?).map(&:name)
+      @columns.values.select { |v| v.accessor }.map(&:name)
     end
 
-    [:string, :text, :integer, :float, :datetime, :date, :boolean, :any].each do |type|
-      define_method(type) do |name, options = {}|
-        @columns << Column.new(name, type, {accessor: @accessors}.merge(options))
+    NO_DEFAULT_GIVEN = Object.new
+    [:string, :text, :integer, :float, :datetime, :date, :boolean, :decimal, :any].each do |type|
+      define_method(type) do |name, **options|
+        if options.key?(:default)
+          options[:default] = decode_default(options[:default])
+        end
+        @columns[name] = Column.new(name, type, options)
       end
     end
-
-    def decimal(name, options = {})
-      @columns << Column.new(name, :decimal, {accessor: @accessors, limit: 20, scale: 6}.merge(options))
+    alias_method :date_time, :datetime
+
+    private
+
+    def decode_default(value)
```

Comment:
@rafaelfranca after you made the [column default to store YAML](https://github.com/kirs/activerecord-typedstore/commit/6b3aea9244ebc3097123bb4e6a6f6dea50b7be3a#diff-00f789e4defafff5d5241b35e6a51d0dR43), we have to decode it here but I feel that my solution is weak. Do you have better ideas?